### PR TITLE
[PP] Ensure loss is visible on console for users

### PR DESCRIPTION
**Pull request by lessw2020**
This is similar in spirit to [PR_944](https://github.com/pytorch/torchtitan/pull/944) (cc @lkhphuc)  but takes a slightly different approach.
Problem - users that default turn on PP training will get -1 for their loss.  This is b/c by default, rank 0 is the only one logged.
However, for *most* PP schedules, the loss is output on the last rank.
Thus, users see -1 for loss and it's a bad/confusing experience.

This PR adds a check to review both the current PP schedule (b/c for VBlocks, loss is returned on 0) and if it is a last rank loss schedule, then it checks that the first rank of the last stage is visible in the LOG_RANK environment variable.
If not, it warns the user, using Red for the warning if color is enabled, and highlights the rank they should add in yellow:
Screenshot 2025-03-07 at 11 51 46 AM

Note that I attempted to then modify the LOG_RANK to add the missing last rank...but it has no effect.  This is b/c the --log_rank_filter passed into torchrun is fixed and thus the env has no effect.
We can fix this by moving to our own filtering via python log filtering (thanks to @d4l3k for this idea) and then it would auto-update.
The tradeoff is that we have to init distributed first (to understand the ranks) meaning that at launch, there's a bit of delay before the first logging.  From there, then NCCL warnings are not suppressed b/c they are emitted from .cpp file vs torchrun filtering controls that...so we get some additional console spam.   
This PR thus sticks to a simple warning with Red highlight (assuming color is on) and provide the user how to fix.

### CODE REVIEW

**Review by H-Huang — APPROVED** [review]
Nice!! Thanks for adding it so quickly
